### PR TITLE
EC transition instruction added

### PR DESCRIPTION
**Pull request by esesyn01**
Signed-off-by: wkawka <wiktor.kawka@3mdeb.com>

### CODE REVIEW

**Comment by esesyn01** [comment]
@TomaszAIR

File: docs/common-coreboot-docs/dasharo_tools_suite.md [hunk 1]
```diff
@@ -162,6 +164,63 @@ fwupdmgr refresh
...
+DTS allows to perform full EC transition from vendor to Dasharo/ec FW.
+
+To perform EC transition make sure you are running DTS version 1.0.1 or higher
+and follow this steps:
+
+* After boot, choose option number 9 to drop to Shell
+* Download BIOS and EC update files. You can download them from
+[there (BIOS)](https://cloud.3mdeb.com/index.php/f/491273) and
+[there (EC)](https://cloud.3mdeb.com/index.php/f/491331)
+> Note: At the time writing this instruction, current highest version is 1.3.0
```
Comment (sulewskiprzemyslaw) [inline docs/common-coreboot-docs/dasharo_tools_suite.md:178]: If the above-mentioned problems will be improved, there's no reason to add that information.

File: docs/common-coreboot-docs/dasharo_tools_suite.md [hunk 2]
```diff
@@ -162,6 +164,63 @@ fwupdmgr refresh
...
+DTS allows to update EC to newer version. To properly update it, follow this
+steps:
+
+* Retrieve information about your current EC
+```bash
+system76_ectool_info
+<TBD>
+```
+* Download EC update file. You can find it
+[there (EC)](https://cloud.3mdeb.com/index.php/f/491331)
```
Comment (sulewskiprzemyslaw) [inline docs/common-coreboot-docs/dasharo_tools_suite.md:208]: I think that at this point we should write something like this: Download the newest version of Embedded Controller firmware.

We should also not put hyperlinks here, but show the command to get the firmware.

File: docs/common-coreboot-docs/dasharo_tools_suite.md [hunk 3]
```diff
@@ -162,6 +164,63 @@ fwupdmgr refresh
...
 
+## Dasharo EC Transition
+
+DTS allows to perform full EC transition from vendor to Dasharo/ec FW.
+
+To perform EC transition make sure you are running DTS version 1.0.1 or higher
+and follow this steps:
+
+* After boot, choose option number 9 to drop to Shell
+* Download BIOS and EC update files. You can download them from
```
[the remark this answers is not in the document]
Comment (TomaszAIR) [reply docs/common-coreboot-docs/dasharo_tools_suite.md]: @sulewskiprzemyslaw script for transition will only work with the release v1.3.0 but you are right, it is better to add commands which needs to be executed on the DTS, fixed. https://github.com/Dasharo/docs/pull/291/commits/876bbe8f5cb9fd74e00055a7536ffeee1ea111b0#diff-b5545f47decb8260688b188363790fc5aef4ea727b9c363329f070d1fa3664a4R176

File: docs/common-coreboot-docs/dasharo_tools_suite.md (hunk 1 above)
Comment (TomaszAIR) [reply docs/common-coreboot-docs/dasharo_tools_suite.md]: Links for download added https://github.com/Dasharo/docs/pull/291/commits/876bbe8f5cb9fd74e00055a7536ffeee1ea111b0#diff-b5545f47decb8260688b188363790fc5aef4ea727b9c363329f070d1fa3664a4R176

File: docs/common-coreboot-docs/dasharo_tools_suite.md [hunk 4]
```diff
@@ -28,6 +28,23 @@ Currently, there are:
 This section provide links and changelogs of DTS CE version started from release
 v1.0.0.
 
+### v1.0.1
+
+#### Images
+
+* [USB bootable DTS CE v1.0.0 image](https://3mdeb.com/open-source-firmware/DTS/v1.0.1/dts-base-image-ce-v1.0.1.wic.gz)
```
Comment (sulewskiprzemyslaw) [inline docs/common-coreboot-docs/dasharo_tools_suite.md:35]: Improper version of image

File: docs/common-coreboot-docs/dasharo_tools_suite.md [hunk 5]
```diff
@@ -162,6 +181,150 @@ fwupdmgr refresh
 fwupdmgr update
 ```
 
+## Dasharo EC Transition
+
+DTS allows to perform full EC transition from vendor to Dasharo/EC FW. For now
```
Comment (sulewskiprzemyslaw) [inline docs/common-coreboot-docs/dasharo_tools_suite.md:186]: Maybe we should use the full terms, i. e. Embedded Controller, firmware? At least the first time we mention something?

File: docs/common-coreboot-docs/dasharo_tools_suite.md [hunk 6]
```diff
@@ -162,6 +181,150 @@ fwupdmgr refresh
...
+To perform EC transition make sure you are running DTS version 1.0.1 or higher
+and follow this steps:
+
+* After boot, choose option number 9 to drop to Shell.
+* Plug in power supply, without it, flashing EC is not possible as losing power
+  may cause in damaged firmware.
+* Download BIOS and EC update files.
+
+```bash
+wget -O firmware.rom https://cloud.3mdeb.com/index.php/s/SKpqSNzfFNY7AbK/download
```
Comment (sulewskiprzemyslaw) [inline docs/common-coreboot-docs/dasharo_tools_suite.md:198]: maybe in this command, we should add an output parameter? During testing, I noticed that downloaded files are saved as `download` without any information about the format of the file.

**Review by sulewskiprzemyslaw — CHANGES_REQUESTED** [review]

File: docs/common-coreboot-docs/dasharo_tools_suite.md (hunk 4 above)
Comment (TomaszAIR) [reply docs/common-coreboot-docs/dasharo_tools_suite.md]: fixed https://github.com/Dasharo/docs

File: docs/common-coreboot-docs/dasharo_tools_suite.md (hunk 5 above)
Comment (TomaszAIR) [reply docs/common-coreboot-docs/dasharo_tools_suite.md]: fixed https://github.com/Dasharo/docs/pull/291/commits/1d137963d289dfceb153d241d870f5174335a96b#diff-b5545f47decb8260688b188363790fc5aef4ea727b9c363329f070d1fa3664a4R186

File: docs/common-coreboot-docs/dasharo_tools_suite.md (hunk 6 above)
Comment (TomaszAIR) [reply docs/common-coreboot-docs/dasharo_tools_suite.md:198]: This is way we use here `-O` flag - this command saves file to `firmware.rom`

**Review by sulewskiprzemyslaw — APPROVED** [review]

File: docs/common-coreboot-docs/dasharo_tools_suite.md [hunk 7]
```diff
@@ -162,6 +181,163 @@ fwupdmgr refresh
...
+
+## Dasharo EC update
+
+DTS allows to update Embedded Controller firmware to the newer version. To
+properly update it, follow these steps:
+
+* Retrieve information about your current EC
+
+  ```bash
+  system76_ectool_info
```
Comment (mkopec) [inline docs/common-coreboot-docs/dasharo_tools_suite.md:285]:
```suggestion
  system76_ectool info
```

File: docs/common-coreboot-docs/dasharo_tools_suite.md [hunk 8]
```diff
@@ -162,6 +181,163 @@ fwupdmgr refresh
...
+  System will shut off in 5 seconds
+  Sync
+  ```
+
+  > Note: this is example output, versions may differ
+
+* Retrieve information about your updated EC
+
+  ```bash
+  system76_ectool_info
```
Comment (mkopec) [inline docs/common-coreboot-docs/dasharo_tools_suite.md:330]:
```suggestion
  system76_ectool info
```

Comment (sulewskiprzemyslaw) [reply docs/common-coreboot-docs/dasharo_tools_suite.md]: @mkopec improved

File: docs/common-coreboot-docs/dasharo_tools_suite.md (hunk 7 above)
Comment (sulewskiprzemyslaw) [reply docs/common-coreboot-docs/dasharo_tools_suite.md]: @mkopec added

File: docs/variants/novacustom_ns5x_7x/releases.md [hunk 9]
```diff
@@ -62,12 +62,14 @@ open EC firmware.
...
+- by using
+    [Dasharo Tools Suite](../../../common-coreboot-docs/dasharo_tools_suite/#dasharo-ec-transition).
 
 EC transition procedure should be performed immediately after flashing the new
 version of the firmware.
 
-Usage firmware v1.3.0 without open EC firmware may cause bricking the system,
-and we are not responsible for such damage.
+Use of the firmware v1.3.0 without open EC firmware may cause bricking the
+system, and we are not responsible for such damage.
```
Comment (mkopec) [inline docs/variants/novacustom_ns5x_7x/releases.md:72]: Has this been tested? It should't be *bricked*, just some functionality will be missing

Comment (mkopec) [reply docs/variants/novacustom_ns5x_7x/releases.md:72]: Maybe let's leave it as is if it hasn't been confirmed not to brick laptops

**Pull request merged**